packages/scrollregion/scrollregion.stories.tsx

```diff
@@ -0,0 +1,166 @@
+
+  useFocusVisible({ scope: focusVisibleRef });
+
+  return (
+    <ScrollRegionContainer containerRef={containerRef} dependency={overflow}>
+      {containerTabIndex => (
+        <div ref={focusVisibleRef} style={{ margin: '2px' }}>
+          <input
+            type="checkbox"
+            id="container-toggle"
+            onChange={e => setOverflow(e.target.checked)}
+          />
+          <label htmlFor="container-toggle">Toggle overflow</label>
+          <StyledCustomFocus ref={containerRef} tabIndex={containerTabIndex}>
+            <p style={{ lineHeight: overflow ? '32px' : 'initial' }}>
+              Turnip greens yarrow ricebean rutabaga endive cauliflower sea lettuce kohlrabi
+              amaranth water spinach avocado daikon napa cabbage asparagus winter purslane kale.
+              Celery potato scallion desert raisin horseradish spinach carrot soko. Lotus root water
+              spinach fennel kombu maize bamboo shoot green bean swiss chard seakale pumpkin.
+            </p>
+          </StyledCustomFocus>
+        </div>
+      )}
+    </ScrollRegionContainer>
+  );
+};
+
+export const Hook = () => {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const focusVisibleRef = useRef<HTMLDivElement>(null);
+  const [overflow, setOverflow] = useState(false);
+  const containerTabIndex = useScrollRegion({ containerRef, dependency: overflow });
+
+  useFocusVisible({ scope: focusVisibleRef });
+
+  return (
+    <div ref={focusVisibleRef} style={{ margin: '2px' }}>
+      <input type="checkbox" id="hook-toggle" onChange={e => setOverflow(e.target.checked)} />
+      <label htmlFor="hook-toggle">Toggle overflow</label>
+      <StyledCustomFocus ref={containerRef} tabIndex={containerTabIndex}>
+        <p style={{ lineHeight: overflow ? '32px' : 'initial' }}>
+          Turnip greens yarrow ricebean rutabaga endive cauliflower sea lettuce kohlrabi amaranth
+          water spinach avocado daikon napa cabbage asparagus winter purslane kale. Celery potato
+          scallion desert raisin horseradish spinach carrot soko. Lotus root water spinach fennel
+          kombu maize bamboo shoot green bean swiss chard seakale pumpkin.
+        </p>
+      </StyledCustomFocus>
+    </div>
+  );
+};
+
+const Content = forwardRef<HTMLDivElement, HTMLAttributes<HTMLDivElement>>(({ ...props }, ref) => {
+  const [hasPadding, setHasPadding] = useState(false);
+  const containerRef = useRef<HTMLDivElement>(null);
+  const mergedRefs = mergeRefs([ref, containerRef]);
+
+  const dependency = useMemo(() => {
+    return [hasPadding, props.children];
+  }, [hasPadding, props.children]);
+
+  const containerTabIndex = useScrollRegion({ containerRef, dependency });
+
+  return (
+    <>
+      <input
+        type="checkbox"
+        id="padding-toggle"
+        style={{ margin: '12px 8px' }}
+        onChange={e => setHasPadding(e.target.checked)}
+      />
+      <label htmlFor="padding-toggle">Toggle padding</label>
```

Comment:
[nit] it would probably be good if there were a space between the label and its checkbox.